common/route/awooter/awooter.cc

```diff
@@ -126,30 +127,17 @@ extern "C" {
     void npnr_log_info(const char *const format) { log_info("%s", format); }
     void npnr_log_error(const char *const format) { log_error("%s", format); }
 
-    uint64_t npnr_belid_null() { return wrap(BelId()); }
-    uint64_t npnr_wireid_null() { return wrap(WireId()); }
-    uint64_t npnr_pipid_null() { return wrap(PipId()); }
-
-    int npnr_context_get_grid_dim_x(const Context *const ctx) { return ctx->getGridDimX(); }
-    int npnr_context_get_grid_dim_y(const Context *const ctx) { return ctx->getGridDimY(); }
-    void npnr_context_bind_bel(Context *ctx, uint64_t bel, CellInfo* cell, PlaceStrength strength) { return ctx->bindBel(unwrap_bel(bel), cell, strength); }
-    void npnr_context_unbind_bel(Context *ctx, uint64_t bel) { return ctx->unbindBel(unwrap_bel(bel)); }
-    bool npnr_context_check_bel_avail(Context *const ctx, uint64_t bel) { return ctx->checkBelAvail(unwrap_bel(bel)); }
-    void npnr_context_bind_wire(Context *ctx, uint64_t wire, NetInfo* net, PlaceStrength strength) { ctx->bindWire(unwrap_wire(wire), net, strength); }
-    void npnr_context_unbind_wire(Context *ctx, uint64_t wire) { ctx->unbindWire(unwrap_wire(wire)); }
-    void npnr_context_bind_pip(Context *ctx, uint64_t pip, NetInfo* net, PlaceStrength strength) { ctx->bindPip(unwrap_pip(pip), net, strength); }
-    void npnr_context_unbind_pip(Context *ctx, uint64_t pip) { ctx->unbindPip(unwrap_pip(pip)); }
-    uint64_t npnr_context_get_pip_src_wire(const Context *const ctx, uint64_t pip) { return wrap(ctx->getPipSrcWire(unwrap_pip(pip))); }
-    uint64_t npnr_context_get_pip_dst_wire(const Context *const ctx, uint64_t pip) { return wrap(ctx->getPipDstWire(unwrap_pip(pip))); }
-    float npnr_context_estimate_delay(const Context *const ctx, uint64_t src, uint64_t dst) { return ctx->getDelayNS(ctx->estimateDelay(unwrap_wire(src), unwrap_wire(dst))); }
-    float npnr_context_get_pip_delay(const Context *const ctx, uint64_t pip) { return ctx->getDelayNS(ctx->getPipDelay(unwrap_pip(pip)).maxDelay()); }
-    float npnr_context_get_wire_delay(const Context *const ctx, uint64_t wire) { return ctx->getDelayNS(ctx->getWireDelay(unwrap_wire(wire)).maxDelay()); }
+    BelId npnr_belid_null() { return BelId(); }
+    WireId npnr_wireid_null() { return WireId(); }
+    PipId npnr_pipid_null() { return PipId(); }
+
+    float npnr_context_estimate_delay(const Context *const ctx, const WireId &const src,  const WireId &const dst) { return ctx->getDelayNS(ctx->estimateDelay(src, dst)); }
```

Comment:
the first `const` in `const Context *` is definitely needed to signify that the object pointed to can't be modified (i.e. the pointer is an immutable reference) but is the `*const` - after the `*` really needed too? the pointer itself is passed by value